src/app/dynamic/enterprise/metering/schedule-config/add-dialog/component.ts

```diff
@@ -0,0 +1,169 @@
+  templateUrl: './template.html',
+})
+export class MeteringScheduleAddDialog implements OnInit, OnDestroy {
+  private readonly _unsubscribe = new Subject<void>();
+  readonly controls = Controls;
+  readonly ScheduleOption = DefaultScheduleOption;
+  form: FormGroup;
+
+  private get _group(): DefaultScheduleOption {
+    return this.form.get(Controls.Group).value;
+  }
+
+  private get _schedule(): string {
+    switch (this._group) {
+      case DefaultScheduleOption.Daily:
+        return DefaultSchedule.Daily;
+      case DefaultScheduleOption.Weekly:
+        return DefaultSchedule.Weekly;
+      case DefaultScheduleOption.Monthly:
+        return DefaultSchedule.Monthly;
+      case DefaultScheduleOption.Custom:
+        return this.form.get(Controls.Schedule).value;
+    }
+  }
+
+  private get _interval(): number {
+    switch (this._group) {
+      case DefaultScheduleOption.Daily:
+        return DefaultScheduleInterval.Daily;
+      case DefaultScheduleOption.Weekly:
+        return DefaultScheduleInterval.Weekly;
+      case DefaultScheduleOption.Monthly:
+        return DefaultScheduleInterval.Monthly;
+      case DefaultScheduleOption.Custom:
+        return this.form.get(Controls.Interval).value;
+    }
+  }
+
+  constructor(
+    private readonly _meteringService: MeteringService,
+    private readonly _matDialogRef: MatDialogRef<MeteringScheduleAddDialog>,
+    private readonly _builder: FormBuilder,
+    private readonly _notificationService: NotificationService,
+    @Inject(MAT_DIALOG_DATA) public data: MeteringScheduleAddDialogConfig
+  ) {}
+
+  ngOnInit(): void {
+    this.form = this._builder.group({
+      [Controls.Name]: this._builder.control('', Validators.required),
+      [Controls.Group]: this._builder.control(DefaultScheduleOption.Daily, Validators.required),
+      [Controls.Schedule]: this._builder.control(''),
+      [Controls.Interval]: this._builder.control('', Validators.min(1)),
+    });
+
+    this.form
+      .get(Controls.Group)
+      .valueChanges.pipe(takeUntil(this._unsubscribe))
+      .subscribe(this._onCustomScheduleChange.bind(this));
+  }
+
+  ngOnDestroy(): void {
+    this._unsubscribe.next();
+    this._unsubscribe.complete();
+  }
+
+  add(): void {
+    this._meteringService
+      .addScheduleConfiguration(this._toMeteringScheduleConfiguration())
+      .pipe(take(1))
+      .subscribe(() => {
+        this._notificationService.success('Added schedule configuration');
+        this._matDialogRef.close(true);
+        this._meteringService.onScheduleConfigurationChange$.next();
+      });
+  }
+
+  private _onCustomScheduleChange(schedule: DefaultScheduleOption) {
+    switch (schedule) {
+      case DefaultScheduleOption.Custom:
+        this._updateFieldValidation(true);
+        break;
+      case DefaultScheduleOption.Daily:
+      case DefaultScheduleOption.Monthly:
+      case DefaultScheduleOption.Weekly:
+        this._updateFieldValidation(false);
+    }
```

Comment:
Can be shortened to `this._updateFieldValidation(schedule === DefaultScheduleOption.Custom)`.